src/js/data/cosmos/index.ts

```diff
@@ -0,0 +1,70 @@
+import { of, throwError } from "rxjs";
+import { map, retry, switchMap } from "rxjs/operators";
+import { makeExecutableSchema } from "graphql-tools";
+import { RequestResponse } from "@dcos/http-service";
+import { CosmosClient, PackageVersionsResponse } from "cosmos-client";
+
+import { Package, PackageSchema } from "#SRC/js/data/cosmos/Package";
+import { PackageVersionSchema } from "#SRC/js/data/cosmos/PackageVersion";
+
+export type GeneralArgs = Partial<PackageQueryArgs>;
+
+export interface PackageQueryArgs {
+  name: string;
+}
+
+function isPackageQueryArgs(args: GeneralArgs): args is PackageQueryArgs {
+  return args.name !== undefined;
```

Comment:
is it ok for name to be `null`?

Reply:
`args.name != null` should be the better check or not?

Reply:
actually it's typed to be a string, so checking whether it's actually a string seems even more reasonable to me.